peer/hostport/hostport.go

```diff
@@ -41,22 +43,25 @@ func Identify(peer string) peer.Identifier {
 
 // NewPeer creates a new hostport.Peer from a hostport.PeerIdentifier, peer.Transport, and peer.Subscriber
 func NewPeer(pid PeerIdentifier, transport peer.Transport) *Peer {
-	return &Peer{
-		PeerIdentifier:   pid,
-		transport:        transport,
-		subscribers:      make(map[peer.Subscriber]struct{}),
-		connectionStatus: peer.Unavailable,
+	p := &Peer{
+		PeerIdentifier: pid,
+		transport:      transport,
+		subscribers:    make(map[peer.Subscriber]struct{}),
 	}
+	p.connectionStatus.Store(int32(peer.Unavailable))
+	return p
 }
 
 // Peer keeps a subscriber to send status updates to it, and the peer.Transport that created it
 type Peer struct {
+	l sync.RWMutex
```

Comment:
nit: I would just name it lock, everything else is spelled out/this is what I prefer to do